### ISSUE TITLE
scikit-learn 1.0+ compatible release on PyPI

### ISSUE THREAD

**Issue by auvipy** [automl/auto-sklearn#1371]
I will be happy to contribute

**Comment by eddiebergman** [comment]
Hello @auvipy

It's on our todo list once we get some other PR's that we are currently working on, if youd like to start, we would be very much appreciative.

Doing so would likely require a decent amount of work, more than just updating the requirements file. It would involve going through the [changelog](https://scikit-learn.org/stable/whats_new/v1.0.html) and updating each component as neccessary and updating any test values that would change. At that point, we have external tests that validate that autosklearn does not degrade performance on automlbenchmark which we would perform externally from github.

@mfeurer, do you ideas of what else would be required?

Best,
Eddie

**Comment by auvipy** [comment]
ok i am also up for coding changes as well

**Comment by eddiebergman** [comment]
Thanks @auvipy,

I'll help guide you along the way and answer questions so feel free to open a PR and continue discussion there!

Best,
Eddie

**Comment by mfeurer** [comment]
Hey @eddiebergman we could actually also try a more gradual path in which we first allow both scikit-learn version in parallel and update the individual components over time?

**Comment by Rattko** [comment]
@mfeurer, @eddiebergman is there any timeline for this issue?

**Comment by auvipy** [comment]
I'm willing to restart the effort

**Comment by eddiebergman** [comment]
Hi @auvipy,

Please see #1677, this update will already include the updated scikit-learn :)

Best,
Eddie

**Comment by auvipy** [comment]
the latest PyPI release still using old scikit-learn 
![image](https://github.com/automl/auto-sklearn/assets/6212603/db778308-7a85-42b5-8f7e-7f026aa75efe)

**Comment by eddiebergman** [comment]
Yes, it's not released yet, just saying that the refactored auto-sklearn will already use the latest scikit learn (once it's released)

**Comment by auvipy** [comment]
oh OK

**Comment by auvipy** [comment]
are they work in progress? shouldn't we consider any external contribution to the the project in the time being?

**Comment by eddiebergman** [comment]
We can, but we would hesitate actually encouraging it, as updating sklearn has already been done in the refactor and so has dropping Python 3.7.

Apologies for having it private as there are reasons outside our control for doing so until it is released. I will update the main issue once there is an update.
